soc/riscv/riscv-privilege/andes_v5/l2_cache.c

```diff
@@ -100,12 +104,15 @@ static int andes_v5_l2c_init(const struct device *dev)
 			/* This SoC doesn't have L2 cache */
 			return -ENODEV;
 		}
+	} else {
+		LOG_ERR("Init failed because syscon driver isn't ready\n");
```

Comment:
I don't fully understand what's going on here. You're just logging "Init failed", but init will continue (you're not returning an error, and even with that, init is not stopped). Second, can you explain why this whole section is optional? Isn't syscon available in some cases?

Reply:
Sorry for the unclear words "failed".  What we want to express is that there "might" be errors. 

The syscon node represents our configurable IP, SMU. In order to fit our other SOC that doesn't have SMU, it must be optional.

Reply:
> Sorry for the unclear words "failed". What we want to express is that there "might" be errors.
> 

What does it mean `there "might" be errors.`?

Reply:
Because not only SMU but also other hardware are configurable for our SOC. In some cases, the initialization will fail. In the other combinations, the initialization can continue.

Reply:
I would probably rephrase this a bit and move to `LOG_DBG`

Reply:
Thanks, I've made this improvement.

Reply:
My question is though, why are you enabling syscon for such cases if you know it's going to fail its initialization and end up printing an init message?

Reply:
Okay, I see!
In cases that fail initialization, the system will trigger the exception.  And we'll prefer to let exceptions happen than just return an error in this scenario.

Reply:
which exception is triggered?

Reply:
It will be a load access fault if the L2 cache doesn't exist.